src/syscalls_new.rs

```diff
@@ -0,0 +1,190 @@
+use crate::callback::CallbackSubscription;
+use crate::callback::SubscribableCallback;
+use crate::shared_memory::SharedMemory;
+use core::isize;
+
+#[derive(Debug)]
+pub enum Error {
+    Fail,           // Generic failure condition
+    Busy,           // Underlying system is busy; retry
+    Already,        // The state requested is already set
+    Off,            // The component is powered down
+    Reserve,        // Reservation required before use
+    Invalid,        // An invalid parameter was passed
+    Size,           // Parameter passed was too large
+    Cancel,         // Operation cancelled by a call
+    Nomem,          // Memory required not available
+    Nosupport,      // Operation or command is unsupported
+    Nodevice,       // Device does not exist
+    Uninstalled,    // Device is not physically installed
+    Noack,          // Packet transmission not acknowledged
+    Unknown(isize), // Unknown Error
+}
+
+struct ReturnCode(isize);
+
+impl From<isize> for ReturnCode {
```

Comment:
I think this implementation should be hidden, s.t. only negative values are accessible. Also, the `ReturnCode` should be more like an `ErrorCode`.